lib/spack/llnl/util/lock.py

```diff
@@ -367,17 +457,57 @@ def release_write(self, release_fn=None):
     def _debug(self, *args):
         tty.debug(*args)
 
-    def _acquired_debug(self, lock_type, wait_time, nattempts):
-        attempts_format = 'attempt' if nattempts == 1 else 'attempt'
-        if nattempts > 1:
-            acquired_attempts_format = ' after {0:0.2f}s and {1:d} {2}'.format(
-                wait_time, nattempts, attempts_format)
-        else:
-            # Dont print anything if we succeeded immediately
-            acquired_attempts_format = ''
-        self._debug(
-            '{0}: {1.path}[{1._start}:{1._length}] [Acquired{2}]'
-            .format(lock_type, self, acquired_attempts_format))
+    def _get_counts_desc(self):
+        return '(reads {0}, writes {1})'.format(self._reads, self._writes) \
+            if tty.is_verbose() else ''
+
+    def _log_acquired(self, lock_type, wait_time, nattempts):
+        attempts_part = get_attempts_str(wait_time, nattempts)
+        now = datetime.now()
+        desc = 'Acquired at %s' % now.strftime("%H:%M:%S.%f")
+        self._debug(self._status_msg(lock_type, '{0}{1}'.
+                                     format(desc, attempts_part)))
+
+    def _log_acquiring(self, lock_type):
+        self._verbose(self._status_msg(lock_type, 'Acquiring'))
+
+    def _log_downgraded(self, wait_time, nattempts):
+        attempts_part = get_attempts_str(wait_time, nattempts)
+        now = datetime.now()
+        desc = 'Downgraded at %s' % now.strftime("%H:%M:%S.%f")
+        self._debug(self._status_msg('READ LOCK', '{0}{1}'
+                                     .format(desc, attempts_part)))
+
+    def _log_downgrading(self):
+        self._verbose(self._status_msg('WRITE LOCK', 'Downgrading'))
+
+    def _log_released(self, lock_type):
+        now = datetime.now()
+        desc = 'Released at %s' % now.strftime("%H:%M:%S.%f")
+        self._debug(self._status_msg(lock_type, desc))
+
+    def _log_releasing(self, lock_type):
+        self._verbose(self._status_msg(lock_type, 'Releasing'))
+
+    def _log_upgraded(self, wait_time, nattempts):
+        attempts_part = get_attempts_str(wait_time, nattempts)
+        now = datetime.now()
+        desc = 'Upgraded at %s' % now.strftime("%H:%M:%S.%f")
+        self._debug(self._status_msg('WRITE LOCK', '{0}{1}'.
+                                     format(desc, attempts_part)))
+
+    def _log_upgrading(self):
+        self._verbose(self._status_msg('READ LOCK', 'Upgrading'))
+
+    def _status_msg(self, lock_type, status):
+        status_desc = '[{0}] {1}'.format(status, self._get_counts_desc())
+        return '{0}{1.desc}: {1.path}[{1._start}:{1._length}] {2}'.format(
+            lock_type, self, status_desc)
+
+    def _verbose(self, *args):
+        # TODO: Is there another level that won't pollute the output for
+        # TODO: tests (e.g., test_spec_json) that could be used?
+        tty.debug(*args)
```

Comment:
Sure.  Apologies for the confusion.  I was using `tty.verbose` here but when I ran tests after a merge (and IIRC without verbose), the output used to test correctness for a few tests, like `test_spec_json`, would contain the messages and fail even though the test should've passed.

I wanted to retain this method since it's a single place to make a change to filter out the extra messages when you're trying to understand what is happening with locks in a large, distributed build.  Bad idea?

Reply:
What would you prefer wrt this issuse @alalazo @tgamblin ?